test/azure/terraform_azure_availabilityset_example_test.go

```diff
@@ -0,0 +1,58 @@
+// +build azure
+
+// NOTE: We use build tags to differentiate azure testing because we currently do not have azure access setup for
+// CircleCI.
+
+package test
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/azure"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTerraformAzureAvailabilitySetExample(t *testing.T) {
+	t.Parallel()
+
+	// subscriptionID is overridden by the environment variable "ARM_SUBSCRIPTION_ID"
+	subscriptionID := ""
+	prefix := "terratest-avs"
```

Comment:
This prefix should use `random.UniqueId()`. This ensures that the test resources all get unique names that won't conflict with other resources in the same Azure account, including other resources from tests running in parallel.

Reply:
Used the native Terraform random_strings resource to generate unique names. The names are collected in outputs, removing the need for the prefix. See main.tf for the implementation. 

https://registry.terraform.io/providers/hashicorp/random/latest/docs/resources/string

Reply:
Please provide feedback on the random_string implementation and which strategy is best:
1. random_string and name concatenation in terraform, values are red from output.
2.  terraform.Options with one prefix using the random.UniqueId() and concatenation using terraform vars "${var.prefix}-network"
3. terraform.Options with one random.UniqueId() per resource name

Of these, the team saw #3 implemented in the azure and is considered the current standard.